Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Task.java

```diff
@@ -396,14 +410,79 @@ public void setUsers(List<User> users) {
 
     /**
      * Set list of user groups.
-     * 
+     *
      * @param userGroups
      *            as list
      */
     public void setUserGroups(List<UserGroup> userGroups) {
         this.userGroups = userGroups;
     }
 
+    /**
+     * Get list of folders whose contents are to be generated.
+     *
+     * @return list of Folder objects or empty list
+     */
+    public List<Folder> getContentFolders() {
+        if (this.contentFolders == null) {
+            this.contentFolders = new ArrayList<>();
+        }
+        return contentFolders;
+    }
+
+    /**
+     * Set list of folders whose contents are to be generated.
+     *
+     * @param contentFolders
+     *            as list
+     */
+    public void setContentFolders(List<Folder> contentFolders) {
+        this.contentFolders = contentFolders;
+    }
+
+    /**
+     * Get list of switch objects for all folders whose contents can be
+     * generated.
+     *
+     * @return list of GeneratorSwitch objects or empty list
+     */
+    @SuppressWarnings("serial")
+    public List<GeneratorSwitch> getGenerators() {
+        if (this.contentFolders == null) {
+            this.contentFolders = new ArrayList<>();
+        }
+        Stream<Project> projects = template.getProjects().stream();
+
+        // Ignore all projects that do not have a source folder configured. It
+        // isn’t possible to generate anything without a data source.
+        Stream<Project> projectsWithSourceFolder = projects.filter(λ -> Objects.nonNull(λ.getGeneratorSource()));
+
+        // Drop all folders to generate if they are their own source folder. The
```

Comment:
please don't litter the code with too much inline comments. The first sentence is enough explanation. Furthermore it is better for code readability to extract functionality in seperate private classes with a narrative name, so you can skip the comments, which is even better!
For example use a method "dropOwnSourceFolders" for the next lines.